astropy/io/fits/file.py

```diff
@@ -105,7 +105,7 @@ class _File:
     """
 
     def __init__(self, fileobj=None, mode=None, memmap=None, overwrite=False,
-                 cache=True):
+                 cache=True, **kwargs):
```

Comment:
for what it's worth, I prefer to avoid `**kwargs` in the signature, spell out all possible kwargs instead. 
(I also very much like to have kwarg only arguments for the optional, but that points out of scope for this PR. So maybe as a middle ground, add the new ones as kwarg only?)

Reply:
In a [previous comment](https://github.com/astropy/astropy/pull/13238#pullrequestreview-974604140), @mhvk suggested trying to avoid adding more arguments to `fits.open()`.  This led me to move all the `fsspec` related code out of `io.fits` and into `utils.data.get_readable_fileobj()` (which was a great idea).  It also led me to remove the `use_fsspec` and `fsspec_kwargs` arguments from the `io.fits` functions (which was perhaps a step too far?)

The two fsspec arguments are passed as follows: `fits.open()` => `fits.HDUList.fromfile()` => `fits._File.__init__()` => `utils.data.get_readable_fileobj()`.  Should we make them explicit in all four functions?

Reply:
I'm all for being explicit everywhere. (and would only use **kwargs in cases where it's a wrapper for significantly different backends, all with a lot of different possible parameters, e.g. Table.read() )

Reply:
I agree.  I'll go ahead and make the arguments explicit in all three fits methods, and add them to the docstrings (where docstrings exist), unless anyone wants to argue otherwise?

Reply:
Yes, this is fine too - explicit is better than implicit. I also like @bsipocz's suggestion to make the new arguments keyword only.

Reply:
Update: I have now made the two new parameters explicit keyword-only arguments. (commit 6e7b6095f97e25d8491cd2ae0d61b2052dc19d3c)